### PR TITLE
[Feature] Add option to generate multiple stories

### PR DESCRIPTION
**Pull request by Ananta2003**
In this commit, we implemented a feature that allows users to generate multiple stories. The following changes were made:

1. Submission Button: A new submission button was created, enabling users to initiate the story generation process seamlessly.
2. Multiple Selection Option: Introduced a multiple select option, allowing users to choose several stories at once for generation. This enhances user flexibility and customization.
3. User Interface Update: The user interface was updated to incorporate the new button and multiple selection feature, ensuring a smooth and intuitive user experience.

Please note that my OpenAI access has expired, so I couldn't actually run the code, but the features were running fine during development. If the implementation doesn't work as expected, feel free to ask for changes. If it does work, feel free to commit the code!

Screenshot 2024-10-03 113411

### CODE REVIEW

**Review by eduardconstantin — CHANGES_REQUESTED** [review]
I can't check the code generation yet but this mostly looks good, just a few things that are not quite right:
- the folders are not highlighted in blue (see pucture)
image
- when I select a folder the arrow disappears and it appears after I press up or down, the first file or folder inside that folder should be highlighted

**Comment by Ananta2003** [comment]
In the UI, the arrow is initially hidden, but it becomes visible when navigating. Could you please provide more details regarding the remaining issues?

**Comment by eduardconstantin** [comment]
> In the UI, the arrow is initially hidden, but it becomes visible when navigating. Could you please provide more details regarding the remaining issues?

That's the issue, the arrow should not be hidden

File: package.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
   },
   "dependencies": {
     "@inquirer/core": "7.0.0",
+    "chalk": "^5.3.0",
```
Comment (eduardconstantin) [inline package.json:37]: we don't need this package
```suggestion
```

File: src/converter.js [hunk 2]
```diff
@@ -1,29 +1,36 @@
...
-      },
-    ],
-    max_tokens: 2048,
-    temperature: 0.5,
-    top_p: 1.0,
-    frequency_penalty: 0.5,
-    presence_penalty: 0.1,
-  });
+  const convertedComponents = [];
+  for (const component of components) {
```
Comment (eduardconstantin) [inline src/converter.js:10]: in index.js you are sending the files one by one in the forEach loop, there is no need for this for loop here

File: src/selector.js [hunk 3]
```diff
@@ -19,18 +19,30 @@ const isSelectableChoice = (file) => {
 export const fileSelector = createPrompt((config, done) => {
   const { basePath = "./", message, pageSize = 10, extensions = [".js", ".jsx", ".ts", ".tsx"] } = config;
 
-  const [cursorPosition, setCursorPos] = useState(1);
+  const [cursorPosition, setCursorPos] = useState(0);
```
Comment (eduardconstantin) [inline src/selector.js:22]: The initial index should be 1
```suggestion
  const [cursorPosition, setCursorPos] = useState(1);
```

File: src/selector.js [hunk 4]
```diff
@@ -19,18 +19,30 @@ const isSelectableChoice = (file) => {
...
   const [filePath, setFilePath] = useState(basePath);
   const [status, setStatus] = useState("pending");
+  const [selectedFiles, setSelectedFiles] = useState([]);
+
+  // Function to reset cursor to the first selectable item
+  const resetCursorToFirstSelectable = (files) => {
```
Comment (eduardconstantin) [inline src/selector.js:28]: You can remove this function

File: src/selector.js [hunk 5]
```diff
@@ -19,18 +19,30 @@ const isSelectableChoice = (file) => {
...
+  // Retrieve files and directories
   let files = fs
     .readdirSync(filePath)
     .reduce((acc, file) => {
       const fullPath = path.join(filePath, file);
       const isDirectory = fs.lstatSync(fullPath).isDirectory();
 
       if (isDirectory || extensions.includes(path.extname(file))) {
-        const displayText = isDirectory ? `\x1b[94m\x1b[1m[DIR]\x1b[0m ${file}` : file;
+        const displayText = isDirectory ? `\x1b[94m\x1b[1m[DIR]${file}\x1b[0m ` : file;
```
Comment (eduardconstantin) [inline src/selector.js:45]:
```suggestion
        const displayText = isDirectory ? `\x1b[94m\x1b[1m[DIR]\x1b[0m ${file}` : file;
```

File: src/selector.js [hunk 6]
```diff
@@ -113,18 +142,18 @@ export const fileSelector = createPrompt((config, done) => {
...
     if (choice.disabled) {
       const disabledLabel = typeof choice.disabled === "string" ? choice.disabled : "(disabled)";
       return `\x1b[90m- ${line} ${disabledLabel}\x1b[0m`;
     }
 
     if (index === cursorPosition) {
-      return line.split(" ").length === 2
-        ? `\x1b[36m❯ ${line.split(" ")[0]}\x1b[0m \x1b[36m${line.split(" ")[1]}\x1b[0m`
-        : `\x1b[36m❯ ${line}\x1b[0m`;
+      return `${selectedMarker}\x1b[36m❯ ${line}\x1b[0m`;
```
Comment (eduardconstantin) [inline src/selector.js:153]:
```suggestion
      return line.split(" ").length === 2
        ? `${selectedMarker}\x1b[36m❯ ${line.split(" ")[0]}\x1b[0m \x1b[36m${line.split(" ")[1]}\x1b[0m`
        : `${selectedMarker}\x1b[36m❯ ${line}\x1b[0m`;
```

File: src/selector.js [hunk 7]
```diff
@@ -50,12 +62,15 @@ export const fileSelector = createPrompt((config, done) => {
       return 0;
     });
 
-  const goBackOption = { name: `\x1b[90m[..]  BACK\x1b[0m`, value: "..", isDirectory: false };
+  // Add navigation options
+  const goBackOption = { name: `\x1b[90m[..]  BACK\x1b[0m`, value: "..", isDirectory: true };
```
Comment (eduardconstantin) [inline src/selector.js:66]:
```suggestion
  const goBackOption = { name: `\x1b[90m[..]  BACK\x1b[0m`, value: "..", isDirectory: false};
```

File: src/converter.js [hunk 8]
```diff
@@ -1,29 +1,36 @@
 import getClient from "./client.js";
 
-export async function componentConverter(component, model, template) {
+export async function componentConverter(components, model, template) {
```
Comment (eduardconstantin) [inline src/converter.js:3]:
```suggestion
export async function componentConverter(component, model, template) {
```

**Review by eduardconstantin — CHANGES_REQUESTED** [review]
I put suggestions in the code.

**Comment by Ananta2003** [comment]
Hi @eduardconstantin ,

Thank you for your suggestions! I’ve committed the changes and updated my PR. Please verify when you get the chance.

I also want to thank you for the updates and guidance throughout this process—it has helped me learn a lot.

**Review by eduardconstantin — APPROVED** [review]
Lovely! Thank you for your help!

**Pull request merged**